cloudpickle/cloudpickle.py

```diff
@@ -849,7 +839,10 @@ def save_file(self, obj):
 
         name = obj.name
 
-        retval = pystringIO.StringIO()
+        if "b" in obj.mode or not PY3:
+            retval = io.BytesIO()
+        else:
+            retval = io.StringIO()
```

Comment:
If I read correctly, this change introduces the ability to pickle files with `mode="rb"` which was not implemented before. Could you please extract this change in a separate pull request with a dedicated new test and a dedicated entry for the new feature in the changelog?

Reply:
I confirm that on master, trying to `cloudpickle.dumps(f)` where `f` is an open file object in `rb` mode indeed yields:

```python-traceback
TypeError: cannot serialize '_io.BufferedReader' object
```

Reply:
Will do, ack.